src/components/googleLogInBtn.jsx

```diff
@@ -29,8 +29,13 @@ class GoogleLogInBtn extends Component {
 
         });
       }
+      else {
+        username = snapshot.val().userName
+        console.log(username)
+      }
+      this.props.signInState(true, googleId, username);
```

Comment:
can u explain this for me

Reply:
so basically, after debugging last night, i found that the last issue was that upon the first render username is unidentified. and the same goes for googleId, and the way we passed !null values was through signInState at the very beginning upon successful login.